lib/revision_table_access_check_filter.js

```diff
@@ -81,6 +81,13 @@ module.exports = function(hyper, req, next, options, specInfo) {
                 }
             });
         }
-        return next(hyper, req);
+        return next(hyper, req)
+        .then(function(res) {
+            if (req.query.redirect === false || mwUtil.isNoCacheRequest(req)) {
+                res.status = 200;
+                delete res.headers.location;
```

Comment:
@d00rman: Consistency with the Varnish response is a compelling reason, imho. We want a request directly to RB to work the same way as it does through Varnish. This matters for third party users or Vagrant.